File: app/src/main/java/by/yahorfralou/plaincalendar/widget/data/database/WidgetDao.java
package by.yahorfralou.plaincalendar.widget.data.database;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import by.yahorfralou.plaincalendar.widget.model.WidgetBean;
import by.yahorfralou.plaincalendar.widget.model.WidgetCalendarBean;
import io.reactivex.Maybe;

@Dao
public interface WidgetDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void saveWidget(WidgetBean widgetBean);

    @Query("SELECT * FROM widgets WHERE id = :widgetId")
    Maybe<WidgetBean> getById(long widgetId);

    @Query("DELETE FROM widget_calendar WHERE widget_id = :widgetId")
    void deleteAllWidgetCalendars(long widgetId);

    @Insert
    void saveWidgetCalendars(List<WidgetCalendarBean> list);
}

File: app/src/main/java/by/offvanhooijdonk/plaincalendar/widget/data/database/WidgetDao.kt
package by.offvanhooijdonk.plaincalendar.widget.data.database

import by.offvanhooijdonk.plaincalendar.widget.model.WidgetCalendarModel
import by.offvanhooijdonk.plaincalendar.widget.model.WidgetModel
import kotlinx.coroutines.flow.Flow

//@Dao
interface WidgetDao {
    //@Insert(onConflict = OnConflictStrategy.REPLACE)
    fun saveWidget(widgetModel: WidgetModel)

    //@Query("SELECT * FROM widgets WHERE id = :widgetId")
    fun getById(widgetId: Long): Flow<WidgetModel?>

    //@Query("DELETE FROM widget_calendar WHERE widget_id = :widgetId")
    fun deleteAllWidgetCalendars(widgetId: Long)

    //@Insert
    fun saveWidgetCalendars(list: List<WidgetCalendarModel>)
}
